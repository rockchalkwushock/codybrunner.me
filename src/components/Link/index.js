import React from 'react'
import styled from 'styled-components'
import Link from 'gatsby-link'
import { bool, string } from 'prop-types'

// Need to do this for testing purposes.
/* istanbul ignore next */
const UnstyledLink = props => <Link {...props} />

const ExternalLink = styled.a`
  color: ${props => props.theme.site.linkClr};
  transition: all 0.2s ease-in;
  :visited {
    color: ${props => props.theme.site.linkClr};
  }
  :hover {
    color: ${props => props.theme.site.linkHvClr};
  }
`

const InternalLink = styled(UnstyledLink)`
  color: ${props => props.theme.site.linkClr};
  transition: all 0.2s ease-in;
  :visited {
    color: ${props => props.theme.site.linkClr};
  }
  :hover {
    color: ${props => props.theme.site.linkHvClr};
  }
`
ExternalLink.displayName = 'ExternalLink'
InternalLink.displayName = 'InternalLink'

const SiteLink = ({ children, ext, href, text }) =>
  ext ? (
    <ExternalLink href={href}>{children || text}</ExternalLink>
  ) : (
    <InternalLink to={href}>{text}</InternalLink>
  )

SiteLink.propTypes = {
  ext: bool,
  href: string.isRequired,
  text: string
}

export default SiteLink
