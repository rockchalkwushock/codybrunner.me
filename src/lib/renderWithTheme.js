import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from './theme'

const renderWithTheme = Component =>
  mount(<ThemeProvider theme={theme}>{Component}</ThemeProvider>)

export default renderWithTheme
