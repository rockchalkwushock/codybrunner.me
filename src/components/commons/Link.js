import React from 'react'
import styled from 'styled-components'
import Link from 'gatsby-link'
import { bool, string } from 'prop-types'

// Need to do this for testing purposes.
/* istanbul ignore next */
const UnstyledLink = props => <Link {...props} />

export const ExternalLink = styled.a`
  color: ${({ theme }) => theme.site.linkClr};
  transition: ${({ theme }) => theme.site.transition};
  :visited {
    color: ${({ theme }) => theme.site.linkClr};
  }
  :hover {
    text-decoration: underline;
  }
`

export const InternalLink = styled(UnstyledLink)`
  color: ${({ theme }) => theme.site.linkClr};
  transition: ${({ theme }) => theme.site.transition};
  :visited {
    color: ${({ theme }) => theme.site.linkClr};
  }
  :hover {
    color: ${({ theme }) => theme.site.linkHvClr};
  }
`
ExternalLink.displayName = 'ExternalLink'
InternalLink.displayName = 'InternalLink'

const SiteLink = ({ children, ext, href, label, text }) =>
  ext ? (
    <ExternalLink href={href} aria-label={label}>
      {children || text}
    </ExternalLink>
  ) : (
    <InternalLink to={href} aria-label={label}>
      {children || text}
    </InternalLink>
  )

SiteLink.propTypes = {
  ext: bool,
  href: string.isRequired,
  label: string,
  text: string
}

export default SiteLink
