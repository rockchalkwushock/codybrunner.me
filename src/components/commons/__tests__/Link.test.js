import React from 'react'
import { shallow } from 'enzyme'

import renderWithTheme from '../../../lib/renderWithTheme'
import Link, { ExternalLink, InternalLink } from '../Link'

describe('Component: <Link />', () => {
  let external
  let internal
  beforeEach(() => {
    external = shallow(<Link ext href="https://cnn.com" text="CNN" />)
    internal = shallow(<Link href="/about" text="About" />)
  })

  test('should render correctly as ExternalLink', () => {
    expect(external).toMatchSnapshot()
  })
  test('should render correctly as InternalLink', () => {
    expect(internal).toMatchSnapshot()
  })
})

describe('Component: <ExternalLink />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = renderWithTheme(
      <ExternalLink href="cnn.com" aria-label="link to cnn" />
    )
  })
  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('should have styles', () => {
    expect(wrapper).toHaveStyleRule('color', '#143a52')
    expect(wrapper).toHaveStyleRule('transition', 'all 0.2s ease-in')
    expect(wrapper).toHaveStyleRule('color', '#143a52', {
      modifier: ':visited'
    })
    expect(wrapper).toHaveStyleRule('color', '#ffe1b6', {
      modifier: ':hover'
    })
  })
})

// FIXME: Issue with `history undefined` coming from HOC.
// I beleive props are not being passed through properly.
describe.skip('Component: <InternalLink />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = renderWithTheme(
      <InternalLink to="/about" aria-label="link about page" />
    )
  })
  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('should have styles', () => {
    expect(wrapper).toHaveStyleRule('color', '#143a52')
    expect(wrapper).toHaveStyleRule('transition', 'all 0.2s ease-in')
    expect(wrapper).toHaveStyleRule('color', '#143a52', {
      modifier: ':visited'
    })
    expect(wrapper).toHaveStyleRule('color', '#ffe1b6', {
      modifier: ':hover'
    })
  })
})
