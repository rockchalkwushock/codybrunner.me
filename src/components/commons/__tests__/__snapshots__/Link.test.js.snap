// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <ExternalLink /> should render correctly 1`] = `
.c0 {
  color: #143a52;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.c0:visited {
  color: #143a52;
}

.c0:hover {
  color: #ffe1b6;
}

<ThemeProvider
  theme={
    Object {
      "post": Object {
        "bg": "#f9f8eb",
        "fg": "",
        "ftClr": "#000",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1.1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#7a9eb1",
        "linkWt": "600",
        "lnHt": "1.618",
        "ltSp": "1px",
      },
      "screen": Object {
        "large": "736px",
      },
      "site": Object {
        "bg": "#7a9eb1",
        "fg": "#ffe1b6",
        "ftClr": "#143a52",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#ffe1b6",
        "lnHt": "1.4",
        "ltSp": "0px",
      },
    }
  }
>
  <ExternalLink
    aria-label="link to cnn"
    href="cnn.com"
  >
    <a
      aria-label="link to cnn"
      className="c0"
      href="cnn.com"
    />
  </ExternalLink>
</ThemeProvider>
`;

exports[`Component: <InternalLink /> should render correctly 1`] = `undefined`;

exports[`Component: <Link /> should render correctly as ExternalLink 1`] = `
<ExternalLink
  href="https://cnn.com"
>
  CNN
</ExternalLink>
`;

exports[`Component: <Link /> should render correctly as InternalLink 1`] = `
<InternalLink
  to="/about"
>
  About
</InternalLink>
`;
